src/Metadata/Resource/ResourceNameCollection.php

```diff
@@ -19,13 +19,15 @@
 final class ResourceNameCollection implements \IteratorAggregate, \Countable
 {
     private $classes;
+    private $context;
 
     /**
      * @param string[] $classes
      */
-    public function __construct(array $classes = [])
+    public function __construct(array $classes = [], array $context = [])
```

Comment:
Yeah, especially that injecting NameCollectionFactory in the MetadataFactory is crap. Thing is I wanted to avoid parsing configuration twice, if you have an idea here I'm all ears!